### PR TITLE
BUG-ID: CLOUDSTACK-8922:  Unable to delete IP tag

### PR DESCRIPTION
**Pull request by yvsubhash**
Root Cause: Due to wrong search string we are fetching a resource which does not belong to current account and domain. This results in permission exception.
Solution: Code changes has been done to fix the incorrect search string.

### CODE REVIEW

**Comment by wilderrodrigues** [comment]
Hi @yvsubhash 

Any details about how you tested it?

Cheers,
Wilder

**Comment by yvsubhash** [comment]
It is tested manually with the data similar to the one mentioned in the bug

**Comment by runseb** [comment]
@yvsubhash we should try to remove manual testing. Can you write a marvin test for it. thanks. cc @wilderrodrigues

**Comment by rajesh-battala** [comment]
Marvin test really helps for validating this bug fix.

**Comment by rhtyd** [comment]
@yvsubhash please rebase against latest master and push -f, update on status of your PR; please add a marvin test that can validate the fix

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 99
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 170
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by niteshsarda** [comment]
I have tested this and **LGTM** for test.

**Comment by yvsubhash** [comment]
tag:mergeready

File: test/integration/component/test_tags.py [hunk 1]
```diff
@@ -2543,3 +2543,248 @@ def test_24_public_IP_tag(self):
             "List tags should return empty response"
         )
         return
+    @attr(tags=["advanced"], required_hardware="false")
```
Comment (DaanHoogland) [inline test/integration/component/test_tags.py]: Do we really need hardware to test this, @yvsubhash ?

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java [hunk 2]
```diff
@@ -297,16 +297,46 @@ public String getUuid(String resourceId, ResourceObjectType resourceType) {
     @ActionEvent(eventType = EventTypes.EVENT_TAGS_DELETE, eventDescription = "deleting resource tags")
     public boolean deleteTags(List<String> resourceIds, ResourceObjectType resourceType, Map<String, String> tags) {
         Account caller = CallContext.current().getCallingAccount();
+        List<String> uuids = new ArrayList<String>();
```
Comment (DaanHoogland) [inline server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: deleteTags() is already to big. Please factor out the new code in a separate set of methods.

File: test/integration/component/test_tags.py (hunk 1 above)
Comment (yvsubhash) [inline test/integration/component/test_tags.py]: @DaanHoogland  it can be run on simulator.  that is the reason required_hardware is set to false. May I now the concern here

Comment (DaanHoogland) [inline test/integration/component/test_tags.py]: I frased my question wrong. can this be moved to the smoke tests?

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 745
 Hypervisor xenserver
 NetworkType Advanced
 Passed=105
 Failed=2
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_list_ids_parameter.py

 * ContextSuite context=TestListIdsParams>:setup Failing since 19 runs

* test_nic.py

 * test_01_nic Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_routers_network_ops.py
test_disk_offerings.py

File: test/integration/component/test_tags.py (hunk 1 above)
Comment (yvsubhash) [inline test/integration/component/test_tags.py]: @DaanHoogland  this test can be moved to smoke tests.

**Comment by karuturi** [comment]
@DaanHoogland Can you review?

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1032
 Hypervisor xenserver
 NetworkType Advanced
 Passed=99
 Failed=16
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_routers_iptables_default_policy.py

 * test_01_single_VPC_iptables_policies Failed

 * test_01_single_VPC_iptables_policies Failing since 2 runs

* test_routers.py

 * test_03_restart_network_cleanup Failing since 11 runs

 * test_04_restart_network_wo_cleanup Failing since 11 runs

 * test_05_router_basic Failing since 11 runs

 * test_06_router_advanced Failing since 11 runs

 * test_08_start_router Failing since 11 runs

 * test_09_reboot_router Failing since 11 runs

* test_network.py

 * test_reboot_router Failing since 11 runs

* test_volumes.py

 * test_06_download_detached_volume Failing since 2 runs

* test_ssvm.py

 * test_10_destroy_cpvm Failed

* test_nic.py

 * test_01_nic Failing since 11 runs

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 119 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 119 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 116 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 116 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_router_dnsservice.py
test_scale_vm.py
test_service_offerings.py
test_loadbalance.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

File: test/integration/component/test_tags.py (hunk 1 above)
Comment (yvsubhash) [reply test/integration/component/test_tags.py]: @DaanHoogland  all the tests in test_tags moved to smoke as they all run on simulator

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1168
 Hypervisor xenserver
 NetworkType Advanced
 Passed=129
 Failed=7
 Skipped=13

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_scale_vm.py

 * ContextSuite context=TestScaleVm>:setup Failing since 40 runs

* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failing since 2 runs

* test_volumes.py

 * test_06_download_detached_volume Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 32 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 159 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 154 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 154 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_04_vpn_tag
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_router_dnsservice.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1176
 Hypervisor xenserver
 NetworkType Advanced
 Passed=132
 Failed=9
 Skipped=41

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_vm_snapshots.py

 * test_change_service_offering_for_vm_with_snapshots Failed

* test_tags.py

 * test_06_template_tag Failed

* test_deploy_vm_iso.py

 * test_deploy_vm_from_iso Failed

* test_volumes.py

 * test_06_download_detached_volume Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 39 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 166 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 161 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 161 runs

 * ContextSuite context=TestRedundantIsolateNetworks>:teardown Failing since 14 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_04_vpn_tag
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_disable_oobm_ha_state_ineligible
test_ha_kvm_host_degraded
test_ha_kvm_host_fencing
test_ha_kvm_host_recovering
test_hostha_configure_default_driver
test_hostha_enable_ha_when_host_disabled
test_hostha_enable_ha_when_host_disconected
test_hostha_enable_ha_when_host_in_maintenance
test_remove_ha_provider_not_possible
test_configure_ha_provider_invalid
test_configure_ha_provider_valid
test_ha_configure_enabledisable_across_clusterzones
test_ha_disable_feature_invalid
test_ha_enable_feature_invalid
test_ha_list_providers
test_ha_multiple_mgmt_server_ownership
test_ha_verify_fsm_available
test_ha_verify_fsm_degraded
test_ha_verify_fsm_fenced
test_ha_verify_fsm_recovering
test_hostha_configure_default_driver
test_hostha_configure_invalid_provider
test_hostha_disable_feature_valid
test_hostha_enable_feature_valid
test_hostha_enable_feature_without_setting_provider
test_list_ha_for_host
test_list_ha_for_host_invalid
test_list_ha_for_host_valid
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_router_dnsservice.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_outofbandmanagement_nestedplugin.py
test_portforwardingrules.py
test_non_contigiousvlan.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_disk_offerings.py

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1185
 Hypervisor xenserver
 NetworkType Advanced
 Passed=138
 Failed=5
 Skipped=41

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_router_dns.py

 * test_router_dns_guestipquery Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 46 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 173 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 168 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 168 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_04_vpn_tag
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_disable_oobm_ha_state_ineligible
test_hostha_configure_default_driver
test_hostha_enable_ha_when_host_disabled
test_hostha_enable_ha_when_host_disconected
test_hostha_enable_ha_when_host_in_maintenance
test_hostha_kvm_host_degraded
test_hostha_kvm_host_fencing
test_hostha_kvm_host_recovering
test_remove_ha_provider_not_possible
test_configure_ha_provider_invalid
test_configure_ha_provider_valid
test_ha_configure_enabledisable_across_clusterzones
test_ha_disable_feature_invalid
test_ha_enable_feature_invalid
test_ha_list_providers
test_ha_multiple_mgmt_server_ownership
test_ha_verify_fsm_available
test_ha_verify_fsm_degraded
test_ha_verify_fsm_fenced
test_ha_verify_fsm_recovering
test_hostha_configure_default_driver
test_hostha_configure_invalid_provider
test_hostha_disable_feature_valid
test_hostha_enable_feature_valid
test_hostha_enable_feature_without_setting_provider
test_list_ha_for_host
test_list_ha_for_host_invalid
test_list_ha_for_host_valid
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_router_dnsservice.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_outofbandmanagement_nestedplugin.py
test_portforwardingrules.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1188
 Hypervisor xenserver
 NetworkType Advanced
 Passed=129
 Failed=8
 Skipped=41

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_scale_vm.py

 * ContextSuite context=TestScaleVm>:setup Failing since 45 runs

* test_portforwardingrules.py

 * test_01_create_delete_portforwarding_fornonvpc Failed

 * test_01_create_delete_portforwarding_fornonvpc Failing since 2 runs

* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failing since 2 runs

* test_list_ids_parameter.py

 * ContextSuite context=TestListIdsParams>:setup Failing since 63 runs

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 49 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 176 runs

 * ContextSuite context=TestRedundantIsolateNetworks>:setup Failing since 12 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_04_vpn_tag
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_disable_oobm_ha_state_ineligible
test_hostha_configure_default_driver
test_hostha_enable_ha_when_host_disabled
test_hostha_enable_ha_when_host_disconected
test_hostha_enable_ha_when_host_in_maintenance
test_hostha_kvm_host_degraded
test_hostha_kvm_host_fencing
test_hostha_kvm_host_recovering
test_remove_ha_provider_not_possible
test_configure_ha_provider_invalid
test_configure_ha_provider_valid
test_ha_configure_enabledisable_across_clusterzones
test_ha_disable_feature_invalid
test_ha_enable_feature_invalid
test_ha_list_providers
test_ha_multiple_mgmt_server_ownership
test_ha_verify_fsm_available
test_ha_verify_fsm_degraded
test_ha_verify_fsm_fenced
test_ha_verify_fsm_recovering
test_hostha_configure_default_driver
test_hostha_configure_invalid_provider
test_hostha_disable_feature_valid
test_hostha_enable_feature_valid
test_hostha_enable_feature_without_setting_provider
test_list_ha_for_host
test_list_ha_for_host_invalid
test_list_ha_for_host_valid
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_router_dnsservice.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_outofbandmanagement_nestedplugin.py
test_login.py
test_deploy_vm_iso.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_disk_offerings.py

**Comment by niteshsarda** [comment]
Tested this fix and it seems to be working as per expectations. **LGTM for test.** Following are the results of the test cases that I have executed locally : 

Test Create tag on LB rule and remove the LB rule ... === TestName: test_01_lbrule_tag | Status : SUCCESS ===
ok
Test Create tag on nat rule and remove the nat rule ... === TestName: test_02_natrule_tag | Status : SUCCESS ===
ok
Test Create tag on firewall rule and remove the firewall rule ... === TestName: test_03_firewallrule_tag | Status : SUCCESS ===
ok
Test Create tag on vpn and remove the vpn ... SKIP: VPN resource tags are unsupported in 4.0
Test creation, listing and deletion tags on UserVM ... === TestName: test_05_vm_tag | Status : SUCCESS ===
ok
Test creation, listing and deletion tag on templates ... === TestName: test_06_template_tag | Status : SUCCESS ===
ok
Test creation, listing and deletion tags on ISO ... === TestName: test_07_iso_tag | Status : SUCCESS ===
ok
Test creation, listing and deletion tagson volume ... === TestName: test_08_volume_tag | Status : SUCCESS ===
ok
Test creation, listing and deletion tag son snapshot ... === TestName: test_09_snapshot_tag | Status : SUCCESS ===
ok
Testcreation, listing and deletion tags on guest network ... === TestName: test_10_network_tag | Status : SUCCESS ===
ok
Test migration of a tagged vm and delete the tag ... === TestName: test_11_migrate_tagged_vm_del | Status : SUCCESS ===
ok
Test to verify that tags are not case sensitive ... === TestName: test_13_tag_case_insensitive | Status : SUCCESS ===
ok
Test multiple tags and with special characters on same machine ... === TestName: test_14_special_char_mutiple_tags | Status : SUCCESS ===
ok
Test creation, listing and deletion tags on projects ... === TestName: test_15_project_tag | Status : SUCCESS ===
ok
Test Query the tags from other account ... === TestName: test_16_query_tags_other_account | Status : SUCCESS ===
ok
Test Query the tags from admin account ... === TestName: test_17_query_tags_admin_account | Status : SUCCESS ===
ok
Test listAPI with invalid tags parameter ... === TestName: test_18_invalid_list_parameters | Status : SUCCESS ===
ok
Test deletion and addition of same tag on a resource. ... === TestName: test_19_delete_add_same_tag | Status : SUCCESS ===
ok
Test creation of same tag on multiple resources ... === TestName: test_20_create_tags_multiple_resources | Status : SUCCESS ===
ok
Test creation of tag on stopped vm. ... === TestName: test_21_create_tag_stopped_vm | Status : SUCCESS ===
ok
Test creation of tag on stopped vm. ... === TestName: test_22_create_tag_destroyed_vm | Status : SUCCESS ===
ok
Hosts without tag are not listed while listing the hosts for migration for instance with tag ... === TestName: test_23_list_untagged_host_for_vm_migration | Status : SUCCESS ===
ok
Testcreation, adding and removing tag on public IP address ... === TestName: test_24_public_IP_tag | Status : SUCCESS ===
ok
Testcreation, adding and removing tag on public IP address ... === TestName: test_25_IP_Address_tag | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 24 tests in 502.723s

OK (SKIP=1)

**Comment by rhtyd** [comment]
LGTM, pending travis results.
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java [hunk 3]
```diff
@@ -296,21 +296,9 @@ public String getUuid(String resourceId, ResourceObjectType resourceType) {
...
-        sb.and("resourceType", sb.entity().getResourceType(), SearchCriteria.Op.EQ);
-
-        SearchCriteria<ResourceTagVO> sc = sb.create();
-        sc.setParameters("resourceId", resourceIds.toArray());
-        sc.setParameters("resourceUuid", resourceIds.toArray());
-        sc.setParameters("resourceType", resourceType);
 
-        List<? extends ResourceTag> resourceTags = _resourceTagDao.search(sc, null);
-        ;
+        Account caller = CallContext.current().getCallingAccount();
```
Comment (rafaelweingartner) [inline server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: This variable called is not used?

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java [hunk 4]
```diff
@@ -360,6 +348,53 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
         return true;
     }
 
+    private List<? extends ResourceTag>  getResourceTags(final List<String> resourceIds, ResourceObjectType resourceType) {
+
+        List<String> uuids = new ArrayList<String>();
+        List<String> internalIds = new ArrayList<String>();
+        for(String resourceId : resourceIds){
+            if(resourceId.matches("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")){
```
Comment (rafaelweingartner) [inline server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: What about simplifying this ?
You can use `org.apache.commons.lang.math.NumberUtils.isDigits(String)`, to check if you have a long there, then you can convert it directly to Long, otherwise, you can check if it empty using `org.apache.commons.lang3.StringUtils.isNotBlank(CharSequence)`, and if it is not blank, you can add in the UUID list.

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java (hunk 3 above)
Comment (yvsubhash) [reply server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: It seems be about caller. that got used in 
 _accountMgr.checkAccess(caller, null, false, owner);

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java [hunk 5]
```diff
@@ -360,6 +348,53 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
         return true;
     }
 
+    private List<? extends ResourceTag>  getResourceTags(final List<String> resourceIds, ResourceObjectType resourceType) {
+
+        List<String> uuids = new ArrayList<String>();
+        List<String> internalIds = new ArrayList<String>();
```
[the remark this answers is not in the document]
Comment (yvsubhash) [reply server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: No, putting internalIds does not help in anyway, as we get are using them again to construct the query

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java (hunk 4 above)
Comment (yvsubhash) [reply server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: yes. But validating uuid also is possibly more foolproof

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java [hunk 6]
```diff
@@ -360,6 +348,53 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
...
+        for(String resourceId : resourceIds){
+            if(resourceId.matches("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")){
+                uuids.add(resourceId);
+            }else{
+                Long internalId = null;
+                try {
+                    internalId = Long.parseLong(resourceId);
```
[the remark this answers is not in the document]
Comment (yvsubhash) [reply server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: @rafaelweingartner  taken care of this

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java [hunk 7]
```diff
@@ -360,6 +348,53 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
...
+                } catch (final NumberFormatException e) {
+                    internalId = null;
```
[the remark this answers is not in the document]
Comment (yvsubhash) [reply server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: @rafaelweingartner  addressed

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (rafaelweingartner) [reply server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: What do you mean by using them to construct query?

File: server/src/com/cloud/tags/TaggedResourceManagerImpl.java (hunk 4 above)
Comment (rafaelweingartner) [reply server/src/com/cloud/tags/TaggedResourceManagerImpl.java]: Well, yes and no. I was thinking about it yesterday, do we need to validate it really?
If it is a string, we could simply use it, if it is invalid, we will not find anything.
I see validation useful when you display a message back to the user saying, "hey, the data you entered is not following the expected format, which should be <formatExpected>...."
That is not what is happening here.

**Comment by bwsw** [comment]
It looks overcomplicated, really, search over resourceId is a wrong logic. It should search only against uuds. We discussed the problem in a mail list with @rafaelweingartner.
Right now I'm working on fix for it too. The code is here:

https://github.com/bwsw/cloudstack/blob/4.10.1-NP/server/src/com/cloud/tags/TaggedResourceManagerImpl.java

All marvin component and smoke are passing, and tag specific too:

```
nosetests-2.7 --with-marvin --marvin-config=setup/dev/advanced.cfg -w test/integration/component --with-xunit --xunit-file=/tmp/bvt_selfservice_cases.xml --zone=Sandbox-simulator --hypervisor=simulator -a tags=advanced,required_hardware=false -a speed=0 `pwd`/test/integration/component/test_tags.py
```
I believe, leaving internal id search is a wrong way to deal with it, because they can not and shouldn't appear in API. Also, the code I'm working on includes security fix for listing resource tags because right now the code from the PR doesn't even checks privileges for listing:

```
 @Override
    public List<? extends ResourceTag> listByResourceTypeAndId(ResourceObjectType type, long resourceId) {
        return _resourceTagDao.listBy(resourceId, type);
    }
```
Mine:

```
@Override
    public List<? extends ResourceTag> listByResourceTypeAndId(ResourceObjectType resourceType, long resourceId) {
        Account caller = CallContext.current().getCallingAccount();
        Pair<Long, Long> accountDomainPair = getAccountDomain(resourceId, resourceType);
        Long domainId = accountDomainPair.second();
        Long accountId = accountDomainPair.first();
        checkResourceAccessible(accountId, domainId, "Account '" + caller + "' doesn't have permissions to list tags" +
                " for resource '" + resourceId + "' of type '" + resourceType + "'.");
        return _resourceTagDao.listBy(resourceId, resourceType);
    }
```

So, basically the PR:

1. it doesn't fix the semantical problem allowing to search with internal ids
2. doesn't fix security fault when listing resource tags.

I need couple of days to finish the code because I want to add account tags feature and have to implement set of marvin tests for it.

**Comment by rafaelweingartner** [comment]
@bwsw I thought this was your PR, sorry for my lack of attention. So it seems someone else also ran into the same problem as you.

Well, if you will still open another PR for this, we can check and discuss both solutions (pros and cons).

**Comment by bwsw** [comment]
@rafaelweingartner looks reasonable

**Comment by bwsw** [comment]
@rafaelweingartner @yvsubhash Take a look at 
https://github.com/apache/cloudstack/pull/2350

**Comment by yvsubhash** [comment]
closing this pr as #2350 takes care of this
